kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs

```diff
@@ -1679,33 +1686,45 @@ impl<'tcx> GotocCtx<'tcx> {
         let sig = self.current_fn().sig();
         let sig = self.tcx.normalize_erasing_late_bound_regions(ty::ParamEnv::reveal_all(), sig);
         // we don't call [codegen_function_sig] because we want to get a bit more metainformation.
+        let is_vtable_shim =
+            matches!(self.current_fn().instance().def, ty::InstanceDef::VTableShim(..));
         let mut params: Vec<Parameter> = sig
             .inputs()
             .iter()
             .enumerate()
-            .map(|(i, t)| {
-                let lc = Local::from_usize(i + 1);
-                let mut ident = self.codegen_var_name(&lc);
-
-                // `spread_arg` indicates that the last argument is tupled
-                // at the LLVM/codegen level, so we need to declare the indivual
-                // components as parameters with a special naming convention
-                // so that we can "retuple" them in the function prelude.
-                // See: compiler/rustc_codegen_llvm/src/gotoc/mod.rs:codegen_function_prelude
-                if let Some(spread) = self.current_fn().mir().spread_arg {
-                    if lc.index() >= spread.index() {
-                        let (name, _) = self.codegen_spread_arg_name(&lc);
-                        ident = name;
+            .filter_map(|(i, t)| {
+                if self.is_zst(*t) && !(i == 0 && is_vtable_shim) {
```

Comment:
I don't think it's worth it. I could split it, but it will just complicate the code flow. We should really be looking at the FnAbi to decide how to codegen parameters.

Reply:
Done